src/test/java/uk/ac/cam/cl/dtg/segue/api/monitors/MisuseMonitorTest.java

```diff
@@ -77,151 +80,116 @@ public final void misuseMonitorTokenOwnerLookup_checkForMisuse_emailShouldBeSent
     replay(this.dummyCommunicator);
 
     for (int i = 0; i < tokenOwnerLookupMisuseHandler.getSoftThreshold(); i++) {
-      try {
-        misuseMonitor.notifyEvent(userId, event);
-
-      } catch (SegueResourceMisuseException e) {
-        fail("Exception should not be thrown after " + tokenOwnerLookupMisuseHandler.getSoftThreshold()
-            + " attempts");
-      }
+      assertDoesNotThrow(() -> misuseMonitor.notifyEvent(userId, event),
+          String.format("Exception should not be thrown after %d attempts",
+              tokenOwnerLookupMisuseHandler.getSoftThreshold()));
     }
 
-    for (int i = TokenOwnerLookupMisuseHandler.SOFT_THRESHOLD; i <= TokenOwnerLookupMisuseHandler.HARD_THRESHOLD + 1;
+    for (int i = tokenOwnerLookupMisuseHandler.getSoftThreshold(); i < tokenOwnerLookupMisuseHandler.getHardThreshold();
          i++) {
-      try {
-        misuseMonitor.notifyEvent(userId, event);
-        if (i >= TokenOwnerLookupMisuseHandler.HARD_THRESHOLD) {
-          fail(String.format("Exception not thrown after %s attempts, over limit of %s!", i,
-              TokenOwnerLookupMisuseHandler.HARD_THRESHOLD));
-        }
-      } catch (SegueResourceMisuseException e) {
-        if (i < TokenOwnerLookupMisuseHandler.HARD_THRESHOLD) {
-          fail(String.format("Exception thrown before %s attempts, under limit of %s!", i,
-              TokenOwnerLookupMisuseHandler.HARD_THRESHOLD));
-        }
-      }
+      assertDoesNotThrow(() -> misuseMonitor.notifyEvent(userId, event),
+          String.format("Exception thrown before %d attempts, under limit of %d!", i,
+              tokenOwnerLookupMisuseHandler.getHardThreshold()));
     }
 
+    assertThrows(SegueResourceMisuseException.class, () -> misuseMonitor.notifyEvent(userId, event),
+        String.format("SegueResourceMisuseException not thrown after %d attempts, over limit of %d!",
```

Comment:
Is this actually right? The message is going to say that the number of attempts and the limit is the same.

Reply:
I retained the messages from the original version when I rewrote these tests and that one could have only been called when the loop variable i was equal to the hard threshold, as the loop it was inside was constrained by `i <= hardThreshold` and the failure state for that message required that `i >= hardThreshold`.
As such, it is consistent with the original version of the test but I agree that the wording is a bit awkward. I can have a go at rewording it if you want?

Reply:
Yeah, I think that now that the code is different, having a more informative message is the right thing to do.